mlflow/utils/proto_json_utils.py

```diff
@@ -99,14 +99,25 @@ def _dataframe_from_json(
     from mlflow.types import DataType
 
     if schema is not None:
-        dtypes = dict(zip(schema.input_names(), schema.pandas_types()))
+        if schema.is_tensor_spec():
+            # The schema can be either:
+            #  - a single tensor: attempt to parse all columns with the same dtype
+            #  - a dictionary of tensors: each column gets the type from equaly named tensor
+            if len(schema.inputs) == 1:
+                dtypes = schema.numpy_types()[0]
+            else:
+                dtypes = dict(zip(schema.input_names(), schema.numpy_types()))
+        else:
+            dtypes = dict(zip(schema.input_names(), schema.pandas_types()))
+
         df = pd.read_json(
             path_or_str, orient=pandas_orient, dtype=dtypes, precise_float=precise_float
         )
-        actual_cols = set(df.columns)
-        for type_, name in zip(schema.input_types(), schema.input_names()):
-            if type_ == DataType.binary and name in actual_cols:
-                df[name] = df[name].map(lambda x: base64.decodebytes(bytes(x, "utf8")))
+        if not schema.is_tensor_spec():
+            actual_cols = set(df.columns)
+            for type_, name in zip(schema.input_types(), schema.input_names()):
+                if type_ == DataType.binary and name in actual_cols:
+                    df[name] = df[name].map(lambda x: base64.decodebytes(bytes(x, "utf8")))
         return df
```

Comment:
Should there a corresponding else that converts it to a numpy input, since we know the signature is tensorSpec.

Reply:
Ignore this one.